buffer/src/main/java/io/netty/buffer/api/internal/ResourceSupport.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api.internal;
+
+import io.netty.buffer.api.Drop;
+import io.netty.buffer.api.Owned;
+import io.netty.buffer.api.Resource;
+import io.netty.buffer.api.Send;
+
+import java.util.Objects;
+
+/**
+ * Internal support class for resources.
+ *
+ * @param <I> The public interface for the resource.
+ * @param <T> The concrete implementation of the resource.
+ */
+public abstract class ResourceSupport<I extends Resource<I>, T extends ResourceSupport<I, T>> implements Resource<I> {
```

Comment:
`@UnstableApi`?

Reply:
This class is already in an internal package.